model/map/src/main/java/org/keycloak/models/map/storage/CriteriaOperator.java

```diff
@@ -106,14 +106,38 @@ private static Object getFirstArrayElement(Object[] value) throws IllegalStateEx
         if (value != null && value.length != 0) {
             throw new IllegalStateException("Invalid argument: " + Arrays.toString(value));
         }
-        return Objects::nonNull;
+        
+        return CriteriaOperator::collectionAwareExists;
+    }
+    
+    private static boolean collectionAwareExists(Object checkedObject) {
+        if (Objects.nonNull(checkedObject)) {
+            if (checkedObject instanceof Collection) {
```

Comment:
This is unnecessary, `instanceof` is `false` if `checkedObject` is `null`

Reply:
Thanks. Updated.